### PR TITLE
Add Arduino Zero board

### PR DESCRIPTION
**Pull request by aabadie**
New open custom node gateway code for [Arduino Zero](https://www.arduino.cc/en/Main/ArduinoBoardZero). It's mainly adapted from the Samr21 as their CPU are quite similar (M0, samd21).

This board has 2 UARTS:
- One is available through the EDBG interface e.g avaiable from USB
- One available on the RX/TX (pins 0 and 1).
  It's possible to use it with an XBee shield, as XBee uses one UART without hardware modification: I tested it and it works.

The CI firmwares were not built agains the riot-upstream repo because it doesn't contain yet the code for supporting this board RIOT.

### CODE REVIEW

**Comment by cladmi** [comment]
Can you put a board available so I can test it ?

**Comment by aabadie** [comment]
> Can you put a board available so I can test it ?

The problem is that I only have 2 of them : one is used in a permanent demo, I use the second one for development and experiment. We ordered of few of them, I'll plug them once they arrive.

File: control_node_serial/Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ CC         ?= gcc
 # VERB = 1
 quiet-cmd = $(if $(VERB),$1,$(if $(2),@echo $2 && $1, @$1))
 
-WARNINGS  += -Werror
+#WARNINGS  += -Werror
```
Comment (aabadie) [inline control_node_serial/Makefile]: I should revert this

File: gateway_code/autotest/autotest.py [hunk 2]
```diff
@@ -118,6 +118,7 @@ def _measures_handler(self, measure_str):
     @staticmethod
     def get_local_mac_addr():
         """ Get eth0 mac address """
+        return "aa:aa:aa:aa:aa:aa"
```
Comment (aabadie) [inline gateway_code/autotest/autotest.py]: not sure how to fix this

Comment (rienafairefr) [reply gateway_code/autotest/autotest.py]: in MAC_CMD 
eth0 => enp0s25 on my machine, but we should have something less fragile

Comment (cladmi) [reply gateway_code/autotest/autotest.py]: You could just take the name of the interface from 'config' or use an env variable.

Comment (aabadie) [reply gateway_code/autotest/autotest.py]: Thanks @cladmi, I'll look into it tomorrow.

**Comment by aabadie** [comment]
Weird, integration tests don't pass any more for this branch rebased on the latest master.

**Comment by rienafairefr** [comment]
local integration tests with m3 board ?

**Comment by aabadie** [comment]
> local integration tests with m3 board ?

no just arduino-zero tests

**Comment by aabadie** [comment]
I didn't test the M3 to be honest

**Comment by rienafairefr** [comment]
Just tested master with an m3, running

    docker run --privileged -v $PWD/tests_utils/cfg_dir:/var/local/config -v /dev:/dev -v $PWD:/iot-lab-gateway -e BOARD_TYPE=m3 -e IOTLAB_GATEWAY_CFG_DIR=/var/local/config iot-lab-gateway-tests tox -e local

works to get the board to auto test etc. If you need just the integration tests and don't care about the rest, use `tox -e local -- -w gateway_code.integration` instead of `tox -e local`

**Comment by rienafairefr** [comment]
on a non-m3 board though, `tox -e local` fails because the IOTLAB_GATEWAY_CFG_DIR is overriden inside the tox environement. i'll send a pr

**Comment by aabadie** [comment]
After fighting a bit with the container, I can say that this PR is working with an Arduino Zero. Here are the steps:
1. checkout this branch
2. build the iot-lab-gateway and iot-lab-gateway-tests docker images
3. write `ARDUINO_ZERO` in `tests_utils/cfg_dir/boad_type`
4. write `arduino-zero-00` in `tests_utils/cfg_dir/hostname`
5. run
```
docker run -t -v /dev/ttyACM0:/dev/ttyON_ARDUINO_ZERO -v $PWD:/home/iot-lab-gateway -v $PWD/tests_utils/cfg_dir:/var/local/config -e BOARD_TYPE=arduino-zero -e CONTROL_NODE_TYPE=no -e HOSTNAME=arduino-zero-00 -e IOTLAB_GATEWAY_CFG_DIR=/var/local/config --privileged iot-lab-gateway-test tox -e test
```

After #19 is merged, step 3 and 4 should become obsolete and step 5 will be simpler.

In step 5, note the `-v /dev/ttyACM0:/dev/ttyON_ARDUINO_ZERO` that maps the host board /dev/tty to the one expected in the container. This is important if one wants the integration tests to work.

**Comment by aabadie** [comment]
@rienafairefr, do you see something preventing this PR from being merged ?

**Review by rienafairefr — APPROVED** [review]

**Comment by aabadie** [comment]
Ok so let's go !

**Comment by rienafairefr** [comment]
Feelds good to close an old PR.

You'll plug down an arduino zero on a gateway as ci resources I presume ?

**Pull request merged**